### ISSUE TITLE
Auto-import adding functionality often adds unnecessary prefixed imports

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/sdk#38326]
First - the auto import adding feature generally seems to be working well and is definitely useful, so thanks! I have noticed a number of specific cases where it could be improved. A large percentage of the time I end up modifying the import that it adds to do it in a different way.

I also have a more general comment that the import prefixes that are provided are not something that any user would use on their own - they are simply `prefixN` always. There are some very simple heuristics that could be used here to greatly improve the prefix (package name?).

Some of the specific cases I have come across are listed below, I will follow up here with more as I remember/come across them, but these impacted me just now:

**I reference something that is already imported but it adds a new prefixed import**

This doesn't always happen but it does sometimes, I don't know why but it is really annoying. Screenshot here:

![unecessary_new_import](https://user-images.githubusercontent.com/984921/64714024-c223d800-d472-11e9-9e25-19e09d62d91a.png)

I already have an import to the suggested library in my file. What is really weird here is for the first 2 characters I type it actually has the correct auto-suggest that won't use a new import, it is only after the 3rd character that it starts suggesting the auto-import.

**I am already importing the library but with specific `show <identifier>` statements.**

Current Behavior: A new import is added with a prefix

Desired Behavior: Add an additional identifier to the existing `show`.

**Comment by scheglov** [comment]
re: import prefixes https://dart-review.googlesource.com/c/sdk/+/116746

**Comment by scheglov** [comment]
Updating `show` combinators - not now.
Maybe in the future.
